includes/managers/controls.php

```diff
@@ -862,6 +875,28 @@ public function remove_control_from_stack( $stack_id, $control_id ) {
 		return true;
 	}
 
+	/**
+	 * Should clear stack cache.
+	 * Whether the CSS requires to clear the controls stack cache.
+	 * @since 3.13.0
+	 * @access protected
+	 * @return bool True if the CSS requires to clear the controls stack cache, False otherwise.
+	 */
+	public function stack_cache_has_been_cleared() {
+		return $this->has_been_cleared;
+	}
+
+	/**
+	 * Clear stack.
+	 * This method clears the stack.
+	 * @since 3.13.0
+	 * @access public
+	 */
+	public function clear_stack_cache() {
+		$this->stacks = [];
+		$this->has_been_cleared = true;
```

Comment:
```suggestion
		$this->has_stacks_cache_been_cleared = true;
```

Reply:
What about the use case where controls are already duplicated when CSS is generated?